### PR TITLE
Orca calc update

### PR DESCRIPTION
**Pull request by gelzinyte**
Updated to be consistent with VASP and Espresso. Also updated the command line interface and corresponding test. Needs #161 to be resolved.

### CODE REVIEW

**Comment by bernstei** [comment]
Any idea why this julia error might be causing the tests to fail?
```
STDERR ERROR: LoadError: MethodError: no method matching pretty_table(::DataFrames.DataFrame, ::Matrix{String}; formatters=PrettyTables.var"#44#46"{Vector{String}}(["%5.3f"]), crop=:horizontal, body_hlines=[3])
```
I'll try to rerun the test job, in case it's something intermittent, but otherwise I don't know why it might have changed from before.

File: tests/calculators/test_orca.py [hunk 1]
```diff
@@ -131,15 +132,15 @@ def test_orca_geometry_optimisation(tmp_path):
...
                 keep_files = "default", 
                 mult=1, 
                 task="opt")
 
 
-    generic.run(inputs=inputs, outputs=outputs, calculator=calc, properties=["energy", "forces"], output_prefix="orca_", num_python_subprocesses=0)
+    generic.run(inputs=inputs, outputs=outputs, calculator=calc, properties=["energy", "forces"], output_prefix="orca_", autopara_info=AutoparaInfo(num_python_subprocesses=0))
 
-    out = [at for at in ConfigSet(outputs)][0]
+    out = [at for at in outputs.to_ConfigSet()][0]
```
Comment (bernstei) [inline tests/calculators/test_orca.py:143]: Any reason why this isn't doing `out = generic.run(...` above? I guess we'll still need an `out = list(out)[0]` statement, but I think it'll still be more readable.

File: tests/calculators/test_orca.py [hunk 2]
```diff
@@ -131,15 +132,15 @@ def test_orca_geometry_optimisation(tmp_path):
...
 
-    calc = ORCA(directory=home_dir, 
+    calc = ORCA(workdir=home_dir, 
                 keep_files = "default", 
                 mult=1, 
                 task="opt")
 
 
-    generic.run(inputs=inputs, outputs=outputs, calculator=calc, properties=["energy", "forces"], output_prefix="orca_", num_python_subprocesses=0)
+    generic.run(inputs=inputs, outputs=outputs, calculator=calc, properties=["energy", "forces"], output_prefix="orca_", autopara_info=AutoparaInfo(num_python_subprocesses=0))
```
Comment (bernstei) [inline tests/calculators/test_orca.py:141]: Why is this intentionally turning off pool autoparallelization? Is it not working with orca?

File: wfl/calculators/orca/basinhopping.py [hunk 3]
```diff
@@ -160,10 +160,10 @@ def __init__(self, atoms=None, n_hop=10, n_run=3, scratch_path=None,
         self.chained_hops = chained_hops
         self.uhf = uhf
 
-        if scratch_path is not None:
-            self.scratch_path = scratch_path
+        if scratchdir is not None:
```
Comment (bernstei) [inline wfl/calculators/orca/basinhopping.py:163]: Do we think that this default should be implemented by `WFLFileIOCalculator`, if possible?

File: wfl/cli/cli.py [hunk 4]
```diff
@@ -460,18 +461,18 @@ def _castep_eval(ctx, inputs, output_file, workdir_root, directory_prefix, prope
...
-def orca_eval(ctx, inputs, workdir_root, output_file, directory_prefix,
-              orca_command, calc_kwargs, keep_files, output_prefix, scratch_path, n_run, n_hop,
+def orca_eval_bh(ctx, inputs, workdir_root, output_file, directory_prefix,
+              orca_command, calc_kwargs, keep_files, output_prefix, scratchdir, n_run, n_hop,
               orca_simple_input, orca_additional_blocks):
     verbose = ctx.obj["verbose"]
 
-    if scratch_path is not None:
-        if not os.path.isdir(scratch_path):
+    if scratchdir is not None:
```
Comment (bernstei) [inline wfl/cli/cli.py:469]: Why is `scratchdir` mandatory here?

Comment (gelzinyte) [reply wfl/cli/cli.py:469]: It isn't mandatory - it defaults to `None` if no argument is given to the command line interface.

File: wfl/calculators/orca/basinhopping.py (hunk 3 above)
Comment (gelzinyte) [reply wfl/calculators/orca/basinhopping.py:163]: This modification came from "replace all". Basin hopping calculator is in need of a larger refactor, but I'd like to discuss with @stenczelt before starting on it.

File: tests/calculators/test_orca.py (hunk 1 above)
Comment (gelzinyte) [reply tests/calculators/test_orca.py:143]: updated!

File: wfl/calculators/orca/basinhopping.py (hunk 3 above)
Comment (bernstei) [reply wfl/calculators/orca/basinhopping.py:163]: OK, let's leave it alone for now.

File: wfl/cli/cli.py (hunk 4 above)
Comment (bernstei) [reply wfl/cli/cli.py:469]: Yeah, I see now. It's just making sure that _if_ it was passed in, it exists.

**Comment by bernstei** [comment]
I'm happy for this to be merged, so @gelzinyte please merge if you're OK with it too.  Did #161 ever get resolved, and we just forgot to close it?

**Comment by gelzinyte** [comment]
Ah forgot about #161, let me add that here

**Comment by gelzinyte** [comment]
ok @bernstei let me know how [cce2463](https://github.com/libAtoms/workflow/pull/162/commits/cce2463251aab036b5c07b7b09f1fd818ed4ce65) looks

**Comment by gelzinyte** [comment]
I think all comments are taken into account?

**Comment by bernstei** [comment]
> I think all comments are taken into account?

Yes. I'll just wait for the tests to finish, and merge when they pass.

**Pull request merged**